specification/desktopvirtualization/resource-manager/Microsoft.DesktopVirtualization/preview/2021-05-13-preview/desktopvirtualization.json

```diff
@@ -3601,6 +3609,283 @@
+        },
+        "credentials": {
+          "$ref": "#/definitions/CredentialsProperties",
+          "description": "Credentials needed to create the virtual machine."
+        }
+      }
+    },
+    "CredentialsProperties": {
+      "description": "Credentials needed to create the virtual machine.",
+      "type": "object",
+      "properties": {
+        "localAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The local admin credentials."
+        },
+        "domainAdmin": {
+          "$ref": "#/definitions/KeyvaultCredentialProperties",
+          "description": "The domain admin credentials."
+        }
+      }
+    },
+    "KeyvaultCredentialProperties": {
+      "description": "Credentials kept in the keyvault.",
+      "type": "object",
+      "properties": {
+        "userName": {
+          "type": "string",
+          "description": "The user name to the account."
+        },
+        "passwordKeyVaultResourceId": {
+          "type": "string",
+          "description": "The keyvault resource id to the keyvault secrets."
+        },
+        "passwordSecretName": {
+          "type": "string",
+          "description": "The keyvault secret name the password is stored in."
+        }
+      }
+    },
+    "SessionHostComponentUpdateConfigurationProperties": {
+      "description": "The session host configuration for updating agent, monitoring agent, and stack component.",
+      "type": "object",
+      "properties": {
+        "maintenanceType": {
+          "type": "string",
+          "enum": [
+            "Default",
+            "TenantAdminControlled"
+          ],
+          "x-ms-enum": {
+            "name": "SessionHostComponentUpdateType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Default",
+                "description": "Agent and other agent side components are delivery schedule is controlled by WVD Infra."
+              },
+              {
+                "value": "TenantAdminControlled",
+                "description": "TenantAdmin have opted in for Scheduled Agent Update feature."
+              }
+            ]
+          }
+        },
+        "useLocalTime": {
+          "type": "boolean",
+          "description": "Whether to use localTime of the virtual machine."
+        },
+        "maintenanceWindowTimeZone": {
+          "type": "string",
+          "description": "Time zone for maintenance as defined in https://docs.microsoft.com/en-us/dotnet/api/system.timezoneinfo.findsystemtimezonebyid?view=net-5.0. Must be set if useLocalTime is true."
+        },
+        "primaryWindow": {
+          "$ref": "#/definitions/MaintenanceWindowProperties",
+          "description": "Primary Window of the maintenance. Maintenance windows are 2 hours long. We try to push agent update in this window first."
```

Comment:
Primary Window of the maintenance. Maintenance windows are 2 hours long. We try to push component update in this window first.

Reply:
done